### PR TITLE
improve cloud info generation

### PR DESCRIPTION
**Pull request by Fedosin**
This commit:
1. Adds better errors messages
2. Starts using errors.As() to check 404 errors since gophercloud returns pointers
3. Returns nil if external network is not set in the config

### CODE REVIEW

File: pkg/asset/installconfig/openstack/validation/cloudinfo.go [hunk 1]
```diff
@@ -106,11 +106,11 @@ func (ci *CloudInfo) getFlavor(flavorName string) (*flavors.Flavor, error) {
 
 func (ci *CloudInfo) getNetwork(networkName string) (*networks.Network, error) {
 	if networkName == "" {
-		return &networks.Network{}, nil
+		return nil, nil
 	}
 	networkID, err := networkutils.IDFromName(ci.clients.networkClient, networkName)
 	if err != nil {
-		if errors.Is(err, gophercloud.ErrResourceNotFound{}) {
+		if _, ok := err.(*gophercloud.ErrResourceNotFound); ok {
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/openstack/validation/cloudinfo.go:113]: this seems backwards... we should be moving towards errors.Is/As for error assertions from interface assertions.

File: pkg/asset/installconfig/openstack/validation/cloudinfo.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (ci *CloudInfo) collectInfo(ic *types.InstallConfig) error {
 func (ci *CloudInfo) getSubnet(subnetID string) (*subnets.Subnet, error) {
 	subnet, err := subnets.Get(ci.clients.networkClient, subnetID).Extract()
 	if err != nil {
-		if errors.Is(err, gophercloud.ErrResourceNotFound{}) {
+		if _, ok := err.(*gophercloud.ErrResourceNotFound); ok {
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/openstack/validation/cloudinfo.go:81]: this seems backwards... we should be moving towards errors.Is/As for error assertions from interface assertions

Comment (Fedosin) [reply pkg/asset/installconfig/openstack/validation/cloudinfo.go]: I changed this to errors.As()

**Pull request merged**